### PR TITLE
Map 426/postcodes io refactor

### PR DESCRIPTION
**Pull request by ev-sc**
Fixes MAP-426

- Renames refs from _2025 to _2024
- Changes the POSTCODES_IO_URL setting

### CODE REVIEW

**Comment by ev-sc** [comment]
@janbaykara @conatus I haven't tested this because I don't have mapped running locally yet

File: hub/management/commands/import_new_constituencies.py [hunk 1]
```diff
@@ -17,9 +17,9 @@ class Command(BaseCommand):
 
     """
     This uses the geopackage from
-      https://pages.mysociety.org/2025-constituencies/datasets/parliament_con_2025/latest
+      https://pages.mysociety.org/2025-constituencies/datasets/parliament_con_2024/latest
```
Comment (conatus) [inline hub/management/commands/import_new_constituencies.py:20]: This feels like this shouldn't be in, as presumably mySociety have retained the paths upstream.

File: hub/management/commands/import_new_constituencies.py [hunk 2]
```diff
@@ -71,9 +71,9 @@ def handle(self, quiet: bool = False, *args, **options):
         constituency_lookup = (
             get_dataset_df(
                 repo_name="2025-constituencies",
-                package_name="parliament_con_2025",
+                package_name="parliament_con_2024",
                 version_name="latest",
-                file_name="parl_constituencies_2025.csv",
+                file_name="parl_constituencies_2024.csv",
```
Comment (conatus) [inline hub/management/commands/import_new_constituencies.py:76]: Same for this, presumably these files have the same name.

File: hub/graphql/extensions/analytics.py [hunk 3]
```diff
@@ -37,5 +27,5 @@ def on_operation(self, *args, **kwargs):
                 if not posthog.disabled:
                     posthog.identify(user.id, {"email": user.email})
                     posthog.capture(user.id, "API request", properties=payload)
-        except Exception as e:
+        except:
```
[the remark this answers is not in the document]
Comment (ev-sc) [reply hub/graphql/extensions/analytics.py]: I just fixed this one 👍

**Comment by conatus** [comment]
@ev-sc 

Makes sense. I think a staging test is better than a local test.

My question is: how does this migrate the data that exists in the production instance across? Presumably internally some models have changed, so there maybe needs to be a migration done?

File: hub/management/commands/import_new_constituencies.py (hunk 1 above)
Comment (ev-sc) [reply hub/management/commands/import_new_constituencies.py:20]: Correct  - changing

File: hub/management/commands/import_new_constituencies.py (hunk 2 above)
Comment (ev-sc) [reply hub/management/commands/import_new_constituencies.py:76]: Fixed

File: hub/analytics.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ def imported_data_count_by_region(self) -> List[RegionCount]:
     def imported_data_count_by_area(
         self,
         # "parliamentary_constituency",
-        # "parliamentary_constituency_2025",
```
[the remark this answers is not in the document]
Comment (ev-sc) [reply hub/analytics.py:34]: Also fixed

File: hub/models.py [hunk 5]
```diff
@@ -1021,8 +1021,8 @@ class GeographyTypes(models.TextChoices):
         WARD = "WARD", "Ward"
         ADMIN_DISTRICT = "ADMIN_DISTRICT", "Council"
         PARLIAMENTARY_CONSTITUENCY = "PARLIAMENTARY_CONSTITUENCY", "Constituency"
-        PARLIAMENTARY_CONSTITUENCY_2025 = (
-            "PARLIAMENTARY_CONSTITUENCY_2025",
+        PARLIAMENTARY_CONSTITUENCY_2024 = (
```
Comment (ev-sc) [inline hub/models.py:1024]: @conatus If we revert this line back to "2025", then I believe that will avoid the need for a migration?

Comment (conatus) [reply hub/models.py:1024]: I think our options are: either go whole hog and write the migration, or restrict the changes to just the way Postcodes.io has the data and keep everything internal the same, to minimise change. Having an internal name for something that differs from its expression is going to be confusing, unless ring fenced.

Feels the former is the better solution and the latter is the for now solution if we don't think writing a migration is sensible,

What do you think?

Comment (janbaykara) [reply hub/models.py:1024]: Whole hog IMO.

**Comment by janbaykara** [comment]
Nice one.

Yep, seems we need a Django migration that updates all DB-stored JSON values that replace `parliamentary_constituency_2025` with `parliamentary_constituency_2024`. Potentially could be done by casting to string? Might be a bit beefy.

- `ExternalDataSource.update_mapping` JSON field
- `GenericData.postcode_data` JSON field
- `ReportMap.display_options` JSON field

**Comment by joaquimds** [comment]
We could also export the database to a file, update the file, then reimport it 😈  although I suspect actually restoring to a database in Render will fail if done in such a hacky way, and we should check that with support.

**Comment by ev-sc** [comment]
Would it not be better to just leave the database values as _2025 and change everything on the query side (to postcodes.io) to _2024.
After next Thursday it's all going to be defunct anyway

**Comment by ev-sc** [comment]
Closing this because we found a devops solution instead.

**Pull request closed without merging**